library/spdm_responder_lib/libspdm_rsp_certificate.c

```diff
@@ -111,6 +113,36 @@ libspdm_return_t libspdm_get_response_certificate(libspdm_context_t *spdm_contex
         length = (uint16_t)(spdm_context->local_context.local_cert_chain_provision_size[slot_id] -
                             offset);
     }
+
+    /* In chunk case, ensure the length request can be served, otherwise return error. */
+    if (libspdm_is_capabilities_flag_supported(spdm_context, false,
+                                               SPDM_GET_CAPABILITIES_REQUEST_FLAGS_CHUNK_CAP,
+                                               SPDM_GET_CAPABILITIES_RESPONSE_FLAGS_CHUNK_CAP)) {
+        LIBSPDM_ASSERT (length + sizeof(spdm_certificate_response_t) <=
+                        spdm_context->local_context.capability.max_spdm_msg_size);
+        if ((spdm_context->connection_info.capability.max_spdm_msg_size != 0) &&
+            (length + sizeof(spdm_certificate_response_t) >
+             spdm_context->connection_info.capability.max_spdm_msg_size)) {
+            LIBSPDM_DEBUG((LIBSPDM_DEBUG_ERROR, "cert size > req max_spdm_msg_size\n"));
+            actual_size = length + sizeof(spdm_certificate_response_t);
+            return libspdm_generate_extended_error_response(
+                spdm_context, SPDM_ERROR_CODE_RESPONSE_TOO_LARGE, 0,
+                sizeof(uint32_t), (uint8_t *)&actual_size, response_size, response);
+        }
+    }
+
+    /* Adjust the length according to max_spdm_msg_size to ensure it can be fit into response buffer. */
+    if (spdm_context->connection_info.capability.max_spdm_msg_size != 0) {
```

Comment:
The specification is explicit that
>If the size of a response message exceeds the size of the MaxSPDMmsgSize of the requesting SPDM endpoint, the responding SPDM endpoint shall respond with an ERROR message of ErrorCode=ResponseTooLarge.

so the Responder shouldn't truncate the response based on the Requester's `MaxSPDMmsgSize`.

Reply:
done. This check is removed.